sdk/eventgrid/eventgrid/src/consumer.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Serializer } from "@azure/core-http";
+import { CloudEvent, EventGridEvent, CustomEventDataDecoder } from "./models";
+import {
+  EventGridEvent as EventGridEventMapper,
+  CloudEvent as CloudEventMapper
+} from "./generated/models/mappers";
+import { parseAndWrap } from "./util";
+import { systemDecoders } from "./systemEventDecoders";
+
+const serializer = new Serializer();
+
+function validateRequiredStringProperty(o: any, propertyName: string): void {
```

Comment:
could we put these validators in a separate file? They seem handy, but not strongly tied to the below class